sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/TransformWithStateExec.scala

```diff
@@ -92,6 +93,35 @@ case class TransformWithStateExec(
     }
   }
 
+  /**
+   * We initialize this processor handle in the driver to run the init function
+   * and fetch the schemas of the state variables initialized in this processor.
+   * @return a new instance of the driver processor handle
+   */
+  private def getDriverProcessorHandle: DriverStatefulProcessorHandleImpl = {
+    val driverProcessorHandle = new DriverStatefulProcessorHandleImpl(timeMode)
+    driverProcessorHandle.setHandleState(StatefulProcessorHandleState.PRE_INIT)
+    statefulProcessor.setHandle(driverProcessorHandle)
+    statefulProcessor.init(outputMode, timeMode)
+    driverProcessorHandle
+  }
+
+  /**
+   * Fetching the columnFamilySchemas from the StatefulProcessorHandle
+   * after init is called.
+   */
+  private def getColFamilySchemas(): Map[String, ColumnFamilySchema] = {
+    val driverProcessorHandle = getDriverProcessorHandle
+    val columnFamilySchemas = driverProcessorHandle.getColumnFamilySchemas
+    closeProcessorHandle()
+    columnFamilySchemas
+  }
+
+  private def closeProcessorHandle(): Unit = {
+    statefulProcessor.close()
```

Comment:
It's not only closing the temporal processor handle, right? Will the statefulProcessor instance be ever reused (in driver, and executor with serialized), or is it clearly isolated? Probably better to make clear with this so that future reader won't be confused.

Reply:
This instance won't ever be used again. Will comment.